src/main/java/com/ipiecoles/java/java350/service/EmployeService.java

```diff
@@ -106,12 +106,12 @@ public void embaucheEmploye(String nom, String prenom, Poste poste, NiveauEtude
      * @throws EmployeException Si le chiffre d'affaire traité ne peut être négatif ou null
      * @throws EntityNotFoundException Si le matricule n'existe pas
      */
-    public void calculPerformanceCommercial(String matricule, Long caTraite, Long objectifCa) throws EmployeException {
+    public void calculPerformanceCommercial(String matricule, long caTraite, long objectifCa) throws EmployeException {
         //Vérification des paramètres d'entrée
-        if(caTraite == null || caTraite < 0){
+        if(caTraite < 0){
             throw new EmployeException("Le chiffre d'affaire traité ne peut être négatif ou null !");
         }
-        if(objectifCa == null || objectifCa < 0){
+        if(objectifCa < 0){
```

Comment:
Aurait dû garder et continuer à tester les conditions << not null >>

Reply:
J'ignore si tu l'as vu, j'ai repris le typage des paramètres Long par long. Ce qui exclut le null. 
Le choix de faire ça était pour réduire la complexité congnitive de cette partie.

Reply:
C'est vrai, j'avais raté cette partie. bien pensé

Reply:
Bien vu, après pour réduire la complexité cognitive, en général, il vaut mieux découper la fonction en plusieurs fonctions.